src/utils/traffic-mesh.js

```diff
@@ -92,6 +97,81 @@ const startForwardProxy = async ({ port, frameworkPort, functionsPort, publishDi
   }
 }
 
+const forwardMessagesToLog = ({ log, subprocess }) => {
+  let currentId = null
+  let spinner = null
+
+  const reset = () => {
+    currentId = null
+    spinner = null
+  }
+
+  rl.createInterface(subprocess.stderr)
+    .on('line', (line) => {
+      let data
+      try {
+        data = JSON.parse(line.trim())
+      } catch (error) {
+        log(NETLIFYDEVERR, 'cannot parse log line as JSON, ignoring:', error)
+        return
+      }
+
+      const { error, id, type } = data
+      switch (type) {
+        case 'bundle:start':
+          currentId = id
+          if (!spinner) {
+            spinner = startSpinner({ text: 'Bundling edge handlers...' })
+          }
+          break
+
+        case 'bundle:success':
+          if (currentId !== id) {
+            return
+          }
+
+          stopSpinner({ spinner, error: false, text: 'Done.' })
+          reset()
+          break
+
+        case 'bundle:fail':
+          if (currentId !== id) {
+            return
+          }
+
+          stopSpinner({
+            spinner,
+            error: true,
+            text: (error && error.msg) || 'Failed bundling Edge Handlers',
+          })
+          log(`${NETLIFYDEVLOG} Change any project file to trigger a re-bundle`)
+          reset()
+          break
+
+        default:
+          log(`${NETLIFYDEVWARN} Unknown mesh-forward event '${type}'`)
+          break
+      }
+    })
+    .on('end', () => {
```

Comment:
Would the `close` event be [better](https://nodejs.org/api/readline.html#readline_event_close)?